### PR TITLE
Remove some keys from development

### PR DESCRIPTION
**Pull request by helicopter88**
- Create a DummyMailer so that in development mode Mailgun is not used
- Generate a random Django secret key when in development
- Create a separate env file for development mode where some secret keys are not set.
  Given that we can remove every other secret key, this allows us to use the vault file only in production mode

List of keys removed:
- Mailgun
- Django
- AWS

### CODE REVIEW

File: src/klee_web/settings.py [hunk 1]
```diff
@@ -22,12 +24,16 @@
...
+
 
 # SECURITY WARNING: don't run with debug turned on in production!
 DEBUG = os.environ.get("DEVELOPMENT") is not None
 
-TEMPLATE_DEBUG = os.environ.get("DEVELOPMENT") is not None
+# If we're in debug mode, generate a random key so that we don't need to provide one
+key = ''.join([random.SystemRandom().choice(string.ascii_letters + string.digits) for _ in range(50)]) if DEBUG else ''
+SECRET_KEY = key if DEBUG else os.environ.get("DJANGO_SECRET_KEY")
+
```
Comment (helicopter88) [inline src/klee_web/settings.py:37]: The change down is not really needed, but fetching twice the environment seems wasteful

**Comment by ccadar** [comment]
Looks good, can you just merge the first two commits into one?

**Comment by helicopter88** [comment]
Done

**Comment by helicopter88** [comment]
After some discussion with @andronat , Slanger is going to be removed and we're just going to use angular, I will clean up this pr accordingly tomorrow

**Comment by ccadar** [comment]
@helicopter88 , this looks ready to merge, right?

**Comment by helicopter88** [comment]
Yes, I will remove pusher in another change as it is kinda big to fit in here

**Comment by ccadar** [comment]
Great, let's merge it then, but you'll need to resolve the conflicts with the mainline first. Thanks.

**Comment by helicopter88** [comment]
Done.

**Pull request merged**